### ISSUE TITLE
Metrics regression: some metrics don't appear in the exporter

### ISSUE THREAD

**Issue by hdevalence** [ZcashFoundation/zebra#1349]
**Version**

`main`

**Description**

After #1330 I noticed a problem where a bunch of metrics didn't show up. I'm not sure if it's connected, or a local misconfiguration on my part.

That change bumped the metrics version in order to pick up a new version of the exporter, because the old version used an old version of hyper that expected to run on the Tokio 0.2 runtime.  The new version builds its own single-threaded runtime for the exporter and launches it on a worker thread.

If we have problems like this, we can roll back to the previous version of the metrics crate, and avoid the Tokio incompatibilities by copying the worker thread code from the current prometheus exporter but using it to launch a single-threaded Tokio 0.2 runtime for the old metrics exporter.  This is probably faster than trynig to spend a lot of time debugging metrics issues right now.

**Comment by teor2345** [comment]
I am also seeing these issues, they seem to affect (some) metrics which are updated within async blocks.

**Comment by hdevalence** [comment]
I think the best fix for this would be to do the rollback sketched above, but I think it might be slightly lower priority than other bugs that block the sync process, since it only affects observability.

**Comment by yaahc** [comment]
I'm betting good money on this being the culprit

```
$ cargo tree -d

metrics v0.12.1
├── zebra-consensus v1.0.0-alpha.0 (/home/jlusby/git/zfnd/zebra/zebra-consensus) (*)
├── zebra-network v1.0.0-alpha.0 (/home/jlusby/git/zfnd/zebra/zebra-network) (*)
└── zebra-state v1.0.0-alpha.0 (/home/jlusby/git/zfnd/zebra/zebra-state) (*)

metrics v0.13.0-alpha.8 (https://github.com/ZcashFoundation/metrics?rev=971133128e5aebe3ad177acffc6154449736cfa2#97113312)
├── metrics-exporter-prometheus v0.1.0-alpha.7 (https://github.com/ZcashFoundation/metrics?rev=971133128e5aebe3ad177acffc6154449736cfa2#97113312) (*)
├── metrics-util v0.4.0-alpha.6 (https://github.com/ZcashFoundation/metrics?rev=971133128e5aebe3ad177acffc6154449736cfa2#97113312) (*)
└── zebrad v1.0.0-alpha.0 (/home/jlusby/git/zfnd/zebra/zebrad)
```

There are actually quite a few duplicates in our dependency tree. I'm gonna start with trying to unify the metrics dependencies and then I'll probably go ahead and clean up as many duplicates as I possibly can.
